homeassistant/components/somfy/cover.py

```diff
@@ -0,0 +1,106 @@
+"""
+Support for Somfy Covers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.somfy/
+"""
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION, \
+    ATTR_TILT_POSITION
+from homeassistant.components.somfy import DOMAIN, SomfyEntity, DEVICES, API
+
+DEPENDENCIES = ['somfy']
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Somfy cover platform."""
+    def get_covers():
+        """Retrieve covers."""
+        from pymfy.api.devices.category import Category
+
+        categories = {Category.ROLLER_SHUTTER.value,
+                      Category.INTERIOR_BLIND.value,
+                      Category.EXTERIOR_BLIND.value}
+
+        devices = hass.data[DOMAIN][DEVICES]
+
+        return [SomfyCover(cover, hass.data[DOMAIN][API]) for cover in
+                devices if
+                categories & set(cover.categories)]
+
+    async_add_entities(await hass.async_add_job(get_covers), True)
+
+
+class SomfyCover(SomfyEntity, CoverDevice):
+    """Representation of a Somfy cover device."""
+
+    def __init__(self, device, api):
+        """Initialize the Somfy device."""
+        from pymfy.api.devices.blind import Blind
+        super().__init__(device, api)
+        self.cover = Blind(self.device, self.api)
```

Comment:
Does this do I/O? Is that why we schedule `get_covers` on the executor?

Reply:
There is no I/O here. `get_covers` filter the devices populated by `update_all_devices`.